include/vast/Analysis/Sarif/Sarif.hpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2024-present, Trail of Bits, Inc.
+
+#pragma once
+
+#ifdef VAST_ENABLE_SARIF
+    #include <vector>
+
+    #include <gap/sarif/sarif.hpp>
+
+namespace vast::analysis::sarif {
+
+    class sarif_analysis
+    {
+      private:
+        std::vector< gap::sarif::result > sarif_results;
+
+      public:
+        inline virtual ~sarif_analysis() = default;
+
+        inline const std::vector< gap::sarif::result > &results() const {
+            return sarif_results;
+        }
+
+      protected:
+        inline std::vector< gap::sarif::result > &results() { return sarif_results; }
```

Comment:
```suggestion
        inline const std::vector< gap::sarif::result >& results() { return sarif_results; }
```

Reply:
If we make the result `const` no pass will be able to add results to it

Reply:
Cannot you add something like `append(gap::sarif)` exposing results this way is weird. You can make `sarif_results` protected and don't have a getter at all.